### ISSUE TITLE
Implement periodic rescan

### ISSUE THREAD

**Issue by synox** [java-operator-sdk/java-operator-sdk#575]
I would like to be able to configure a resyncPeriod, as it is possible to configure on https://github.com/kubernetes-client/java/blob/master/util/src/main/java/io/kubernetes/client/informer/SharedInformerFactory.java#L120 in the k8s java client. 

One use case could be to reconcile an external resource (lets say database) once per hour, as we can't listen to changes to it. 

Let me know if there is a workaround.

**Comment by csviri** [comment]
Hi @synox , thx for the question.
you can create an informer like:
```
kubernetesClient.informers().sharedIndexInformerFor(ConfigMap.class, [resync period]);
```

There is a constructor for InformerEventSource:
```
  public InformerEventSource(SharedInformer<T> sharedInformer,
      Function<T, Set<String>> resourceToUIDs) {
    this(sharedInformer, resourceToUIDs, null, true);
  }
```

where this can be passed.

Is this what you need / good enough? :)

**Comment by csviri** [comment]
But another way you can do this is just schedule an execution of the controller. See the `UpdateControl.reSchedule` on master. (Not released that.). For you purpose would be much clear. 

Also what you can do, if you don't want to execute the whole reconciliation loop, you can implement a custom `EventSource` that will poll the the DB locally, and produce an event when needed. This might be not trivial though.

**Comment by synox** [comment]
Thanks a lot @csviri  for your support. 
I found a usage example for the informer: https://github.com/java-operator-sdk/java-operator-sdk/blob/master/operator-framework/src/test/java/io/javaoperatorsdk/operator/sample/informereventsource/InformerEventSourceTestCustomResourceController.java#L38

`UpdateControl.withReSchedule` seems perfect. 

For both options, I will need to wait a while until this becomes available in quarkus-operator-sdk. 

Meanwhile I found this workaround, until `UpdateControl.withReSchedule`  is available. 

In the Controller: 
```java
    @Override
    public void init(EventSourceManager eventSourceManager) {
        timerEventSource = new TimerEventSource();
        eventSourceManager.registerEventSource("timer", timerEventSource);
    }
```

and then in the event Handling: 
```java
timerEventSource.scheduleOnce(esp, Duration.ofHours(1).toMillis());
return UpdateControl.noUpdate();
```
